app/Http/Controllers/Littercoin/LittercoinController.php

```diff
@@ -24,16 +33,255 @@ public function getUsersLittercoin ()
     }
 
     /**
-     * Sign the transaction
-     *
-     * Submit the transaction
-     *
-     * Update the Littercoin as sent
+     * Get the amount Ada and littercoin in circulation and
+     * the littercoin script address and UTXO with the thread
+     * token in it.
+     */
+    public function getLittercoinInfo ()
+    {
+        //set user as admin for testing
+        //$user = Auth::user();
+        //$user->assignRole('admin');
+
+        $cmd = '(cd ../littercoin/;node get-lc-info.mjs) 2>> ../storage/logs/littercoin.log'; 
+        $response = exec($cmd);
+
+        return [
+            $response
+        ];
+    }
+
+
+    /**
+     * Create the littercoin mint transaction.
+     */
+    public function mintTx (Request $request)
+    {
+        // TODO santize inputs
+        $destAddr = $request->input('destAddr');
+        $changeAddr = $request->input('changeAddr');
+        $utxos = $request->input('utxos');
+        $strUtxos=implode(",",$utxos);
+
+        $userId = Auth::user()->id;
+        $littercoinPaid = Auth::user()->littercoin_paid;
+        $littercoinEarned = Littercoin::where('user_id', $userId)->count();
+        $littercoinDue = $littercoinEarned - $littercoinPaid;
+
+        if ($littercoinDue > 0) {
+            $cmd = '(cd ../littercoin/;node create-lc-mint-tx.mjs '.$littercoinDue.' '.$destAddr.' '.$changeAddr.' '.$strUtxos.') 2>> ../storage/logs/littercoin.log'; 
+            $response = exec($cmd);
+    
+            return [
+                $response
+            ];   
+        } else {
+            return [
+                '{"status": "400", "msg": "Littercoin due must be greater than zero"}'
+            ];
+        }
+    }
+
+    /**
+     * Submit the littercoin mint transaction which includes signing
+     * the tx with a private key.
+     */
+    public function submitMintTx (Request $request)
+    {
+        // TODO santize inputs
+        $cborSig = $request->input('cborSig');
+        $cborTx = $request->input('cborTx');
+
+        $cmd = '(cd ../littercoin/;node submit-lc-mint-tx.mjs '.$cborSig.' '.$cborTx.') 2>> ../storage/logs/littercoin.log'; 
+        $response = exec($cmd);
+        $responseJSON = json_decode($response, false);
+
+        if ($responseJSON->status == 200) {
+
+            // Update the amount of littercoin paid to user in the DB
+
+            // Commented out for testing purposes
+            //$user->littercoin_paid = $littercoinPaid + $littercoinDue;
+            //$user->save();
```

Comment:
We should probably update each littercoin in the new table as sent

Reply:
this has been completed in recent commits